src/Channels/ReferenceCountedBuffer.cs

```diff
@@ -28,19 +25,16 @@ public IBuffer Preserve(int offset, int count, out int newStart, out int newEnd)
 
         public void Dispose()
         {
-            lock (_lockObj)
```

Comment:
This was originally an interlocked but I believe the decrement and dispose need to be atomic.